mautic/mautic
# Fix: The decision step ignores the preceding condition in the campaign

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | Y
| New feature/enhancement? (use the a.x branch)      | N
| Deprecations?                          | N
| BC breaks? (use the c.x branch)        | N
| Automated tests included?              | Y 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
If a conditional step is placed before a decision step in a Mautic campaign, it will be ignored when a negative decision path is executed.



#### Steps to replicate this issue (before applying this PR):

1. Create campaign with decision step after condition, for example: ![obraz](https://user-images.githubusercontent.com/8580942/171001064-1d1f0fd7-6b5b-4d1a-a263-f4a13a300f2d.png)
2. Add contact and run this campaign
3. The condition will be ignored and both paths from example will be executed ![obraz](https://user-images.githubusercontent.com/8580942/171001106-2efb4a37-817e-43c8-bf0e-b0c35aa068ce.png)

#### Steps to test this PR:
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Repeat steps to replicate (1 and 2)
3. Only one path after condition should be executed




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11200"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
